From 3da586e6aee0a58e8df21d310089f639f9ed296d Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 25 Jan 2020 18:31:19 +0200
Subject: [PATCH] [SW-0] hl: preboot: load U-Boot instead of SPL

Since SPL left without any real task beside loading U-Boot
Removing it altogether

Change-Id: I36cc735a1fc04bbf8f726f37ef6cb4b9fabf0eb9
---

diff --git a/app/preboot.c b/app/preboot.c
index 197674e..2616853 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -77,23 +77,20 @@
 	return modifiable_image_offset(pending);
 }
 
-static void preboot_copy_spl(void)
+static void preboot_copy_uboot(void)
 {
-	char *name = "SPL";
+	char *name = "U-Boot";
 	bool upgrade_pending = false;
 	u32 img_offset = next_image_offset(&upgrade_pending);
-	u32 spl_offset = img_offset + HL_SPI_SPL_OFF;
 	u32 uboot_offset = img_offset + HL_SPI_UBOOT_OFF;
-	char *spl_image = upgrade_pending ? "candidate" : "stable";
+	char *uboot_image = upgrade_pending ? "candidate" : "stable";
 
-	preboot_puts_raw("Preparing to run SPL ");
-	preboot_puts_raw(spl_image);
+	preboot_puts_raw("Preparing to run U-Boot ");
+	preboot_puts_raw(uboot_image);
 	preboot_puts_hex_raw(" (offset: ", img_offset, false);
 	preboot_puts(")\n");
 
-	spi_copy((void *)SPL_TEXT_BASE, spl_offset, SPL_SIZE, name);
-
-	regmap_writel(uboot_offset, mmPSOC_GLOBAL_CONF_SCRATCHPAD_30);
+	spi_copy((void *)UBOOT_TEXT_BASE, uboot_offset, UBOOT_SIZE, name);
 }
 
 #ifndef TARGET_IS_TESTER
@@ -140,7 +137,7 @@
 
 void preboot_main(void)
 {
-	void (*spl_start)(void) = (void *)SPL_TEXT_BASE;
+	void (*uboot_start)(void) = (void *)UBOOT_TEXT_BASE;
 
 	regmap_writel(CPU_BOOT_STATUS_IN_PREBOOT, get_cpu_boot_status_reg());
 
@@ -149,7 +146,7 @@
 	hl_pll_init();
 	/* reinit uart after PLLs configured */
 	hl_uart_init();
-	preboot_copy_spl();
+	preboot_copy_uboot();
 
-	spl_start();
+	uboot_start();
 }
diff --git a/include/preboot.h b/include/preboot.h
index ce48b77..293333f 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -15,25 +15,24 @@
 #define HL_SPI_BASE				0x1A000000
 #define HL_CFG_BASE				0x1C000000
 #define SDRAM_BASE				0x10000000
-#define SPL_TEXT_BASE				0x10080000
+#define UBOOT_TEXT_BASE				0x10100000
 #endif
 
 #ifdef __aarch64__
 #define HL_SPI_BASE				0x7FF8000000ull
 #define HL_CFG_BASE				0x7FFC000000ull
 #define SDRAM_BASE				0x7FF0000000ull
-#define SPL_TEXT_BASE				0x7FF0080000ull
+#define UBOOT_TEXT_BASE				0x7FF0100000ull
 #endif
 
 #define PPBOOT_SIZE		0x4000		/* 16KB */
 #define PREBOOT_SIZE		0xC000		/* 48KB */
 
-#define SPL_SIZE		0x10000		/* 64KB */
+#define UBOOT_SIZE		0xF0000		/* 960KB */
 #define HL_UART_BAUDRATE	115200
 #define HL_SPI_FLASH_MAX_FREQ	25000000
 #define HL_SPI_NEXT_IMAGE_OFF	0x60000
 #define HL_SPI_MODIFI_IMAGE_OFF	0x60004
-#define HL_SPI_SPL_OFF		0xA0000
 #define HL_SPI_UBOOT_OFF	0xB0000
 #define HL_SPI_PREBOOT_OFF	0x1A0000
 #define HL_SPI_PRIMARY_BASE	0
